bpf/lib/common.h

```diff
@@ -188,6 +188,19 @@ struct policy_entry {
 	__u64		bytes;
 };
 
+struct data_metrics_key {
+    __u8      reason;     //0: forwarded, >0 dropped
+    __u8      dir:1,      //0: Unknown 1: ingress 2: egress
```

Comment:
This doesn't make sense, `2` uses 2 bits. When is the direction unknown? We can just use a single bit for ingress/egress.

Reply:
Also, do you think 0 or unknown direction is not needed?

Reply:
Changing the number of bits to 2. Please let me know if you feel otherwise.

Reply:
I think it's important to define what this "ingress" / "egress" means. I assume that "ingress" here is describing traffic traversing to an endpoint, and egress is traversing from an endpoint.

"Unknown" to me suggests that the traffic is received to the node, but we don't intend to forward it to an endpoint inside the node. I believe that in all other cases, the traffic should either be clearly being sent by an endpoint, or being sent towards an endpoint. 

That said, I'm not sure whether there are currently any cases inside the BPF code where we are receiving traffic from outside the node and handle forwarding to another node. AFAIK such cases are handled instead by regular linux stack constructs today, such as the route table, etc.

Reply:
@joestringer @tgraf : What will the direction be for`cilium_dbg_capture(skb, DBG_CAPTURE_DELIVERY, skb->ifindex)`  in  `icmp6_send_reply`. and `arp_respond`. I think it should be EGRESS.
Similarly, going through bpf_overlay.c, `cilium_dbg_capture(skb, DBG_CAPTURE_DELIVERY, skb->ifindex);` in this case should both be egress, while in bpf_lb.c its ingress correct?

Reply:
done